kauri/tree.go

```diff
@@ -0,0 +1,170 @@
+package kauri
+
+import (
+	"math"
+
+	"github.com/relab/hotstuff"
+)
+
+// MaxChild determines the branching factor of the tree.
+const MaxChild = 2
+
+// TreeConfiguration is an abstraction for a tree communication model.
+type TreeConfiguration interface {
+	InitializeWithPIDs(ids map[hotstuff.ID]int)
+	GetHeight() int
+	GetChildren() []hotstuff.ID
+	GetSubTreeNodes() []hotstuff.ID
+	GetParent() (hotstuff.ID, bool)
+}
+
+// FaultFreeTree implements a fault free tree configuration.
+type FaultFreeTree struct {
+	ID                  hotstuff.ID
```

Comment:
If I understand correctly this `ID` field is the current/local tree node. If so, perhaps we should use a more clear name.

Reply:
Also, maybe it shouldn't be exported.